chia/farmer/farmer_api.py

```diff
@@ -399,8 +399,17 @@ async def request_signed_values(self, full_node_request: farmer_protocol.Request
             [full_node_request.foliage_block_data_hash, full_node_request.foliage_transaction_block_hash],
         )
 
-        msg = make_msg(ProtocolMessageTypes.request_signatures, request)
-        await self.farmer.server.send_to_specific([msg], node_id)
+        response = await self.farmer.server.call_api_of_specific(HarvesterAPI.request_signatures, request, node_id)
+        if response is None or not isinstance(response, harvester_protocol.RespondSignatures):
+            self.farmer.log.error(f"Invalid response from harvester {node_id} for request_signatures: {response}")
+            return None
+
+        signed_values = self._process_respond_signatures(response)
```

Comment:
maybe adding an additional comment explaining why you separated this function would be nice.